src/storage-client/src/util/remap_handle.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Reclocking compatibility code until the whole ingestion pipeline is transformed to native
+//! timestamps
+
+use timely::progress::frontier::Antichain;
+use timely::progress::Timestamp;
+
+use mz_persist_client::Upper;
+use mz_repr::Diff;
+
+/// A handle that can be used to durably persist a remap collection translating FromTime to
+/// IntoTime.
+#[async_trait::async_trait(?Send)]
+pub trait RemapHandle {
```

Comment:
You can reshape this trait to basically just have the `next()` method. The important thing is that the logic of `next()` remains intact for partitioned sources

Reply:
Did this in a way that let me share the trait between the migration and the reclocking code. However, the `ReclockOperator` is still generic over `Handle`, which means we still need this trait; wasn't sure if your suggestion was ultimately to remove the genericism fom `Handle`, so did not undertake that effort.